crates/federated-server/src/server/state.rs

```diff
@@ -12,6 +14,9 @@ pub struct ServerStateInner<R: engine::Runtime, SR> {
     /// The server runtime, defining how to trigger IO depending on the platform.
     #[cfg_attr(not(feature = "lambda"), allow(unused))]
     pub server_runtime: SR,
+
+    /// For MCP connection management.
+    pub sse_txs: Arc<DashMap<Arc<str>, mpsc::Sender<ClientJsonRpcMessage>>>,
```

Comment:
Would move it in a dedicated `McpState` or equivalent struct that itself is part of the `ServerState` to separate concerns.